File: sprint3/By languages/C++/lesson4code1.cpp
#include <iostream>
using namespace std;

void gen_binary(int n, string prefix) {
    if (n == 0) {
        cout << prefix << endl;
    } else {
        gen_binary(n - 1, prefix + "0");
        gen_binary(n - 1, prefix + "1");
    }
}

int main() {
    gen_binary(3, "");
    return 0;
}

File: sprint3/By languages/JS/lesson4code1.js
function genBinary(n, prefix) {
    if (n == 0) {
        console.log(prefix);
    } else {
        genBinary(n - 1, prefix + "0");
        genBinary(n - 1, prefix + "1");
    }
}

genBinary(3, "");
